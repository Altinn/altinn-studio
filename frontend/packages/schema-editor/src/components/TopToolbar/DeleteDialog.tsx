import React from 'react';
import classes from './DeleteDialog.module.css';
import type { PopoverProps } from '@digdir/design-system-react';
import { Button, ButtonColor, ButtonVariant, Popover } from '@digdir/design-system-react';
import { Panel, PanelVariant } from '@altinn/altinn-design-system';
import { useTranslation } from 'react-i18next';

export type IDeleteDialogProps = {
  schemaName: string;
  onConfirm: () => void;
  onCancel: () => void;
} & Pick<PopoverProps, 'trigger' | 'open'>;

export function DeleteDialog({
  onCancel,
  onConfirm,
  open,
  schemaName,
  trigger,
}: IDeleteDialogProps) {
  const { t } = useTranslation();
  const description = t('administration.delete_model_confirm', { schemaName });
  return (
    <Popover trigger={trigger} open={open} placement='bottom' className={classes.popover}>
      <Panel variant={PanelVariant.Warning}>
        <p>{description}</p>
        <div className={classes.buttons}>
          <Button
            color={ButtonColor.Danger}
            id='confirm-delete-button'
            onClick={onConfirm}
            variant={ButtonVariant.Filled}
          >
            {t('general.continue')}
          </Button>
          <Button
            color={ButtonColor.Secondary}
            id='cancel-delete-button'
            onClick={onCancel}
            variant={ButtonVariant.Filled}
          >
            {t('general.cancel')}
          </Button>
        </div>
      </Panel>
    </Popover>
  );
}
