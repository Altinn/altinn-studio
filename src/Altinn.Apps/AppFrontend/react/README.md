## Running the tests

### Lint checks

1. Navigate to the folder `src/Altinn.Apps/AppFrontend/react`.
2. Execute `npm ci`. This step is only nescessary if you have not already done it, or if you change branches.
3. Execute `npm run install-deps`. This step is only nescessary if you have not already done it, or if you change branches.
4. Execute `npm run lint`.

### Unit tests

1. Navigate to the folder `src/Altinn.Apps/AppFrontend/react`.
2. Execute `npm ci`. This step is only nescessary if you have not already done it, or if you change branches.
3. Execute `npm run install-deps`. This step is only nescessary if you have not already done it, or if you change branches.
4. Execute `npm run test`.
