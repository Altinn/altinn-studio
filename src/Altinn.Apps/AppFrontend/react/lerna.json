{
  "command": {
    "bootstrap": {
      "hoist": true
    }
  },
  "packages": [
    "./*"
  ],
  "version": "independent"
}
