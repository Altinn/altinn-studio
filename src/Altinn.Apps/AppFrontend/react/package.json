{
	"name": "app-frontend",
	"private": true,
	"scripts": {
		"install-deps": "lerna bootstrap --hoist --ci",
		"no-ci-hoist": "lerna bootstrap --hoist --no-ci",
		"install-deps-with-lock": "lerna bootstrap",
		"test": "lerna exec --concurrency 1 -- npm test",
		"install-all": "lerna exec -- npm ci",
		"clean-lock": "lerna exec -- rm ./package-lock.json",
		"clean-node-modules": "rm -rf ./node_modules && lerna exec --concurrency 1 -- rm -rf ./node_modules",
		"update-packages": "lerna exec --concurrency 1 -- npm update --dev",
		"list-outdated": "lerna run list-outdated"
	},
	"devDependencies": {
		"@typescript-eslint/eslint-plugin": "~4.8.2",
		"@typescript-eslint/parser": "~4.8.2",
		"eslint": "~7.14.0",
		"eslint-config-airbnb": "~18.2.1",
		"eslint-plugin-import": "~2.22.1",
		"eslint-plugin-jsx-a11y": "~6.4.1",
		"eslint-plugin-react": "~7.21.5",
		"eslint-plugin-react-hooks": "^4.2.0",
		"ini": "1.3.5",
		"lerna": "3.22.1"
	}
}
