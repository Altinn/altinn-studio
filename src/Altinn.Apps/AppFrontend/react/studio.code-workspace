{
	"folders": [
		{
			"path": "."
		}
  ],
  "settings": {
    "typescript.surveys.enabled": false,
    "typescript.tsdk": "node_modules/typescript/lib",
    "eslint.alwaysShowStatus": true,
    "eslint.validate": [
      "javascript",
      "javascriptreact",
      { "language": "typescript", "autoFix": true },
      { "language": "typescriptreact", "autoFix": true}
    ]
  }
}
