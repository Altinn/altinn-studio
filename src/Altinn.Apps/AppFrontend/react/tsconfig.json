{
  "compilerOptions": {
    "outDir": "compiled",
    "module": "commonjs",
    "target": "es5",
    "lib": [
      "es6",
      "dom",
      "es7",
      "esnext"
    ],
    "sourceMap": true,
    "allowJs": true,
    "jsx": "react",
    "moduleResolution": "node",
    "rootDir": "./",
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "strictNullChecks": false,
    "noUnusedLocals": true
  },
  "include": [
    "**/*.ts",
    "**/*.tsx",
  ],
  "exclude": [
    "node_modules",
    "build",
    "scripts",
    "acceptance-tests",
    "webpack",
    "jest",
    "setupTests.ts",
    "templates",
    "**/setupTests.ts"
  ],
}
