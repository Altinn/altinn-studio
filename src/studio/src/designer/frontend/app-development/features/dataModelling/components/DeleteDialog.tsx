import * as React from 'react';
import Popover from 'app-shared/components/AltinnPopover';
import { getLanguageFromKey, getParsedLanguageFromKey } from 'app-shared/utils/language'

interface IDeleteDialogProps {
  anchor: any,
  language: any,
  schemaName: any,
  onConfirm: any;
  onCancel: any
}

export default function DeleteDialog(props: IDeleteDialogProps) {
  const description = getParsedLanguageFromKey(
    'administration.delete_model_confirm', props.language, [props.schemaName], true,
  );
  return (<Popover
    anchorEl={props.anchor}
    anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
    btnCancelText={getLanguageFromKey('general.cancel', props.language)}
    descriptionText={description}
    btnConfirmText={getLanguageFromKey('general.continue', props.language)}
    btnPrimaryId='confirm-delete-button'
    btnClick={props.onConfirm}
    handleClose={props.onCancel}
    transformOrigin={{ horizontal: 'right', vertical: 'top' }}
  />);
}
