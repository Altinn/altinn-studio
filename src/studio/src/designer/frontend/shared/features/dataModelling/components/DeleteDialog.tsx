import * as React from 'react';
import Popover from '../../../components/AltinnPopover';
import { getLanguageFromKey, getParsedLanguageFromKey } from '../../../utils/language';

interface IDeleteDialogProps {
  anchor: Element,
  language: any,
  schemaName: string,
  onConfirm: () => void,
  onCancel: () => void,
}

export default function DeleteDialog(props: IDeleteDialogProps) {
  const description = getParsedLanguageFromKey(
    'administration.delete_model_confirm', props.language, [props.schemaName], true,
  );
  return (<Popover
    anchorEl={props.anchor}
    anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
    btnCancelText={getLanguageFromKey('general.cancel', props.language)}
    descriptionText={description}
    btnConfirmText={getLanguageFromKey('general.continue', props.language)}
    btnPrimaryId='confirm-delete-button'
    btnClick={props.onConfirm}
    handleClose={props.onCancel}
    transformOrigin={{ horizontal: 'right', vertical: 'top' }}
  />);
}
