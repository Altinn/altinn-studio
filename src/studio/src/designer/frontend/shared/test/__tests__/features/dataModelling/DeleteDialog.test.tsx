import { mount } from 'enzyme';
import 'jest';
import * as React from 'react';
import DeleteDialog from '../../../../features/dataModelling/components/DeleteDialog';

describe('>>> DeleteDialog.tsx', () => {
  const language = { administration: {} };
  let schemaName: string;
  beforeEach(() => {
    schemaName = 'some-name';
  });
  const mountComponent = () => mount(<DeleteDialog
    anchor={document.body}
    language={language}
    schemaName={schemaName}
    onCancel={() => {}}
    onConfirm={() => {}}
  />);
  it('+++ Should match snapshot with the least amount of params', () => {
    const rendered = mountComponent().html();
    expect(rendered).toMatchSnapshot();
  });
});
