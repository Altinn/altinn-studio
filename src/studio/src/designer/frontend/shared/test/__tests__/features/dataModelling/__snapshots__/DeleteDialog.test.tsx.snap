// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> DeleteDialog.tsx +++ Should match snapshot with the least amount of params 1`] = `"<div role=\\"presentation\\" class=\\"MuiPopover-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div aria-hidden=\\"true\\" style=\\"z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: transparent;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiPaper-root MuiPopover-paper MuiPaper-elevation8\\" style=\\"opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;\\" tabindex=\\"-1\\"><div class=\\"MuiGrid-root makeStyles-popover-10 MuiGrid-container MuiGrid-direction-xs-column\\"><p class=\\"MuiTypography-root makeStyles-subHeader-9 MuiTypography-body1\\">administration.delete_model_confirm</p><div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonCommon-2 makeStyles-buttonConfirm-4 MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"button\\" id=\\"confirm-delete-button\\"><span class=\\"MuiButton-label\\">general.continue</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonCommon-2 makeStyles-buttonCancel-3 MuiButton-textPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\"><span class=\\"makeStyles-borderBottom-1\\">general.cancel</span></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;
